apps/dotcom/src/components/IFrameProtector.tsx

```diff
@@ -72,24 +86,25 @@ export function IFrameProtector({
 
 			if (event.data === EXPECTED_RESPONSE) {
 				// todo: check the origin?
-				setEmbeddedState('iframe-ok')
+				setEmbeddedState(EMBEDDED_STATE.IFRAME_OK)
 				clearTimeout(timeout)
 			}
 		}
 
 		window.addEventListener('message', handleMessageEvent, false)
 
-		if (embeddedState === 'iframe-unknown') {
+		if (embeddedState === EMBEDDED_STATE.IFRAME_UNKNOWN) {
 			// We iframe embeddings on multiplayer or readonly
 			if (WHITELIST_CONTEXT.includes(context)) {
 				window.parent.postMessage(EXPECTED_QUESTION, '*') // todo: send to a specific origin?
 				timeout = setTimeout(() => {
-					setEmbeddedState('iframe-not-allowed')
-					trackAnalyticsEvent('connect_to_room_in_iframe', { slug, context })
+					setEmbeddedState(EMBEDDED_STATE.IFRAME_NOT_ALLOWED)
+					const referrer = document.referrer
```

Comment:
`ancestorOrigins` is a more reliable way to checking who's embedding us, referrer might be different if e.g. tldraw is linked to within an iframe: https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins

Reply:
Unfortunately [not supported in Firefox](https://developer.mozilla.org/en-US/docs/Web/API/Location/ancestorOrigins#browser_compatibility). We could fall back to referrer just for Firefox though.

Reply:
ok, added this in (with fallback)